invalid `BeginDocumentationCommentWithOneLineSummary` warning

```swift
  /// Open a scope enclosed by `"typewriter"` double-quotes.
  ///
  /// Closed by a ``EndTypewriterDoubleQuotes`` macro.
  ///
  /// __Example Usage__:
  ///   ```swift
  ///   BeginTypewriterDoubleQuotes()
  ///   "Hello, Swift!"
  ///   EndTypewriterDoubleQuotes()
  ///   ```
  public struct BeginTypewriterDoubleQuotes: MDocMacroProtocol {
    public static let kind = "Qo"
    public var arguments: [MDocASTNode]
    /// Creates a new `BeginTypewriterDoubleQuotes` macro.
    public init() {
      self.arguments = []
    }
  }
```

triggers: `Tools/generate-manual/MDoc/MDocMacro.swift:1144:3: warning: [BeginDocumentationCommentWithOneLineSummary] terminate this sentence with a period: "Open a scope enclosed by `"typewriter"` double-quotes."`

--- a/Sources/SwiftFormat/Rules/BeginDocumentationCommentWithOneLineSummary.swift
+++ b/Sources/SwiftFormat/Rules/BeginDocumentationCommentWithOneLineSummary.swift
@@ -97,8 +97,13 @@ public final class BeginDocumentationCommentWithOneLineSummary: SyntaxLintRule {
     else { return }
 
     // For the purposes of checking the sentence structure of the comment, we can operate on the
-    // plain text; we don't need any of the styling.
-    let trimmedText = briefSummary.plainText.trimmingCharacters(in: .whitespacesAndNewlines)
+    // plain text; we don't need any of the styling. Additionally, the backticks that are
+    // frequently used to denote symbol names cause issues with quote identification, and
+    // aren't necessary for this purpose.
+    let trimmedText = briefSummary.plainText
+      .trimmingCharacters(in: .whitespacesAndNewlines)
+      .replacingOccurrences(of: "``.+?``", with: "ZZZ", options: .regularExpression)
+      .replacingOccurrences(of: "`.+?`", with: "zzz", options: .regularExpression)
     let (commentSentences, trailingText) = sentences(in: trimmedText)
     if commentSentences.count == 0 {
       diagnose(.terminateSentenceWithPeriod(trimmedText), on: decl)

--- a/Sources/SwiftFormat/Rules/BeginDocumentationCommentWithOneLineSummary.swift
+++ b/Sources/SwiftFormat/Rules/BeginDocumentationCommentWithOneLineSummary.swift
@@ -11,8 +11,8 @@
 //===----------------------------------------------------------------------===//
 
 import Foundation
-import SwiftSyntax
 import Markdown
+import SwiftSyntax
 
 #if os(macOS)
 import NaturalLanguage

--- a/Sources/SwiftFormat/Rules/BeginDocumentationCommentWithOneLineSummary.swift
+++ b/Sources/SwiftFormat/Rules/BeginDocumentationCommentWithOneLineSummary.swift
@@ -12,6 +12,7 @@
 
 import Foundation
 import SwiftSyntax
+import Markdown
 
 #if os(macOS)
 import NaturalLanguage
@@ -91,19 +92,20 @@ public final class BeginDocumentationCommentWithOneLineSummary: SyntaxLintRule {
 
   /// Diagnose documentation comments that don't start with one sentence summary.
   private func diagnoseDocComments(in decl: DeclSyntax) {
+    // Extract the summary from a documentation comment, if it exists, and strip
+    // out any inline code segments (which shouldn't be considered when looking
+    // for the end of a sentence).
+    var inlineCodeRemover = InlineCodeRemover()
     guard
       let docComment = DocumentationComment(extractedFrom: decl),
-      let briefSummary = docComment.briefSummary
+      let briefSummary = docComment.briefSummary,
+      let noInlineCodeSummary = inlineCodeRemover.visit(briefSummary) as? Paragraph
     else { return }
 
     // For the purposes of checking the sentence structure of the comment, we can operate on the
-    // plain text; we don't need any of the styling. Additionally, the backticks that are
-    // frequently used to denote symbol names cause issues with quote identification, and
-    // aren't necessary for this purpose.
-    let trimmedText = briefSummary.plainText
+    // plain text; we don't need any of the styling.
+    let trimmedText = noInlineCodeSummary.plainText
       .trimmingCharacters(in: .whitespacesAndNewlines)
-      .replacingOccurrences(of: "``.+?``", with: "ZZZ", options: .regularExpression)
-      .replacingOccurrences(of: "`.+?`", with: "zzz", options: .regularExpression)
     let (commentSentences, trailingText) = sentences(in: trimmedText)
     if commentSentences.count == 0 {
       diagnose(.terminateSentenceWithPeriod(trimmedText), on: decl)
@@ -234,5 +236,11 @@ extension Finding.Message {
     _ text: Sentence
   ) -> Finding.Message {
     "add a blank comment line after this sentence: \"\(text)\""
+  }
+}
+
+struct InlineCodeRemover: MarkupRewriter {
+  mutating func visitInlineCode(_ inlineCode: InlineCode) -> Markup? {
+    nil
   }
 }

--- a/Tests/SwiftFormatTests/Rules/BeginDocumentationCommentWithOneLineSummaryTests.swift
+++ b/Tests/SwiftFormatTests/Rules/BeginDocumentationCommentWithOneLineSummaryTests.swift
@@ -31,6 +31,13 @@ final class BeginDocumentationCommentWithOneLineSummaryTests: LintOrFormatRuleTe
         // ...
         }
 
+        /// Open a scope enclosed by `"typewriter"` double-quotes.
+        struct BeginTypewriterDoubleQuotes: MDocMacroProtocol {}
+
+        /// Before parsing arguments, capture all inputs that follow the `--`
+        /// terminator in this argument array.
+        public static var postTerminator: ArgumentArrayParsingStrategy
+
         /// This docline should not succeed.
         /// There are two sentences without a blank line between them.
         1️⃣struct Test {}

--- a/Tests/SwiftFormatTests/Rules/BeginDocumentationCommentWithOneLineSummaryTests.swift
+++ b/Tests/SwiftFormatTests/Rules/BeginDocumentationCommentWithOneLineSummaryTests.swift
@@ -38,6 +38,10 @@ final class BeginDocumentationCommentWithOneLineSummaryTests: LintOrFormatRuleTe
         /// terminator in this argument array.
         public static var postTerminator: ArgumentArrayParsingStrategy
 
+        /// The source converter for the text in the current test, which
+        /// is related to ``thisInvalid(method.Signature:)``.
+        var converter: SourceLocationConverter!
+
         /// This docline should not succeed.
         /// There are two sentences without a blank line between them.
         1️⃣struct Test {}

